File: 140/Intermediate.clj
; abbreviating functions
(require '[clojure.string :as str])
(def rs read-string)

; makes a vector of n 0s
(defn make-row [n] (into [] (for [x (range n)] (str 0)) ))

(defn intermediate-140 []
  ; setup
  (def matrix {})
  (def settings (str/split (read-line) #" "))
  (def N (rs (first settings)))
  (def M (rs (last settings)))
  ; create matrix
  (doseq [x (range N)]
    (def matrix 
      (assoc matrix x (make-row N))))
  ; populate adjlist
  (doseq [x (range M)]
    (def line (str/split (read-line) #" "))
    (def nodes (split-at (.indexOf line "->") line))
    (doseq [l (first nodes)]
      (doseq [r (rest (last nodes))]
        (def matrix 
          (assoc matrix (rs l) 
            (assoc (matrix (rs l)) (rs r) "1"))))))
  ; printing
  (doseq [x (range N)]
    (prn (str/join (matrix x)))))

(intermediate-140)

File: 140/Intermediate.py
N, M = map(int, raw_input().split())
matrix = [[0 for y in xrange(N)] for x in xrange(N)]
for y in xrange(M):
  line = raw_input()
  lhs = map(int, line[:line.find("->")].split())
  adjs = map(int, line[line.find("->")+2:].split())
  for row in lhs:
    for adj in adjs:
      matrix[row][adj] = 1
print "\n".join("".join(str(el) for el in row)for row in matrix)
